docs/rfc/0004_component_groups_scale_view.md

```diff
@@ -0,0 +1,300 @@
+|                    |                                                         |
+| -------            |---------------------------------------------------------|
+| Feature            | Expose the scale subresource with `RisingWaveScaleView` |
+| Status             | In-progress                                             |
+| Date               | 2022-09-05                                              |
+| Authors            | arkbriar                                                |
+| RFC PR #           |                                                         |
+| Implementation PR #|                                                         |
+|                    |                                                         |
+
+# **Summary**
+
+The RFC aims to provide a new CR to support the scale subresource over groups of some user-specified component of
+the RisingWave resource. With that, the users can benefit from the scale subresource and apply any tools that have such
+requirement. For example, the horizontal auto-scaling service (HPA/[KEDA](https://github.com/kedacore/keda)) will work
+after this is implemented.
+
+# **Motivation**
+
+> Why are we doing this?
+
+The [scale subresource](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource)
+is
+a subresource like the status but exposes three fixed fields:
+
+- `.spec.replicas` which refers to the desired replicas
+- `.status.replicas` which refers to the current replicas
+- `.status.selector` which refers to the Pod selector (but in a serialized manner)
+
+However, the current design of the `RisingWave` resource can not provide such a subresource because it is consisted of
+four independent components, and each of the component can have one or more groups. Therefore, the `kubectl scale` can
```

Comment:
`Group` is a concept proposed and implemented in the current `RisingWave` CRD. Each component (meta/frontend/compute/compactor) could declare several Pod groups, where each group can have its individual resources and scheduling constraints. So we can have a multi-region deployment by declaring two groups for the compute node. 

Every group finally corresponds to a workload object, e.g., Deployment. And the scale view here tries to combine these groups inside a scale subresource so that the auto-scaler would scale them at the same time.

Reply:
Got it thanks!